server/methods.coffee

```diff
@@ -67,14 +67,15 @@ Meteor.methods
           featureClass: geoname.featureClass
           featureCode: geoname.featureCode
           alternateNames: geoname.alternateNames
-      locationAnnotations = locationAnnotations.filter (loc)->
+      locationAnnotations.forEach (loc)->
         geoname = geonamesById[loc.geoname.geonameid]
         if geoname
           loc.geoname = geoname
```

Comment:
That's right. The original version is replaced by setting it to the look-up version stored in the geonamesById map.